File: problems/greedy_algorithms/task_assignment/python/task_assignment.py
def task_assignment(k: int, tasks: list[int]) -> list[list[int]]:
    # pair each task value with its original index
    task_pairs: list[tuple[int, int]] = [
        (task, index) for index, task in enumerate(tasks)
    ]
    # sort the tasks by their values
    task_pairs.sort(key=lambda x: x[0])

    # pair the smallest and largest tasks
    result: list[list[int]] = []
    for i in range(k):
        result.append([task_pairs[i][1], task_pairs[-(i + 1)][1]])

    return result

File: problems/greedy_algorithms/task_assignment/js/task_assignment.js
export const taskAssignment = (k, tasks) => {
  // pair each task value with its original index
  const taskPairs = tasks.map((task, index) => [task, index]);

  // sort the tasks by their values
  taskPairs.sort((a, b) => a[0] - b[0]);

  // pair the smallest and largest tasks
  const result = [];
  for (let i = 0; i < k; i++) {
    result.push([taskPairs[i][1], taskPairs[taskPairs.length - 1 - i][1]]);
  }

  return result;
};
